### PR TITLE
[TE] RCA Advanced Dimension Analysis - Service and Component

### PR DESCRIPTION
**Pull request by newsummit**
This PR is mainly setting up the data flow for Advanced Dimension Analysis table in RCA and includes:
- Ember data wiring: model/adapter/serializer for `dimensions`
- Query caching service for dimensions
- Component for `rootcause-dimensions` (parses data from several RCA objects, including `context` as the user changes the selected metric or analysis window)
- Removing call to `services/all`, which is not needed in `entity-mapping-modal`

Testing:
- Added initial integration test for dimensions component. Skipping for now until I can work out why its not getting mirage data.

331 tests completed in 40442 milliseconds, with 0 failed, 3 skipped, and 0 todo.
427 assertions of 427 passed, 0 failed.

### CODE REVIEW

File: thirdeye/thirdeye-frontend/app/pods/components/modals/entity-mapping-modal/component.js [hunk 1]
```diff
@@ -37,7 +37,7 @@ import _ from 'lodash';
 const MAPPING_TYPES = [
   'METRIC',
   'DIMENSION',
-  'SERVICE',
+  // 'SERVICE', // TODO: Enable once we want to fix services endpoint
```
Comment (aaronucsd) [inline thirdeye/thirdeye-frontend/app/pods/components/modals/entity-mapping-modal/component.js]: Not sure i understand this comment. Is the service not working? We should just fix it.

File: thirdeye/thirdeye-frontend/app/pods/components/rootcause-dimensions/component.js [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+import { get, set, getProperties } from '@ember/object';
+import { toBaselineRange } from 'thirdeye-frontend/utils/rca-utils';
+import { groupedHeaders, baseColumns } from 'thirdeye-frontend/shared/dimensionAnalysisTableConfig';
+
+export default Component.extend({
+  classNames: ['rootcause-dimensions-table'],
+  dimensionsApiService: service('services/api/dimensions'),
+
+  /**
+   * Incoming cached state for rootcause view
```
[the remark this answers is not in the document]
Comment (newsummit) [reply thirdeye/thirdeye-frontend/app/pods/components/rootcause-dimensions/component.js]: Sure. As explained by Alex, the `context` object is maintained by the RCA caching services. It defines the 'state' of the analysis view as the user modifies options. This object will always contain start/end ranges, which the component needs to react to. I can describe this better inline.

File: thirdeye/thirdeye-frontend/app/pods/manage/alerts/performance/route.js [hunk 3]
```diff
@@ -246,7 +246,11 @@ export default Route.extend({
       .then((richFunctionObjects) => {
```
[the remark this answers is not in the document]
Comment (newsummit) [reply thirdeye/thirdeye-frontend/app/pods/manage/alerts/performance/route.js]: Sure can. I just haven't refactored yet. Have been wanting to go through last year's code to replace all nested then structures.

File: thirdeye/thirdeye-frontend/app/pods/services/api/dimensions/service.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+import EmberObject, { computed, get } from '@ember/object';
+
+/**
+ * @type {Ember.Service}
+ * @summary This service provides all the api calls for dimensions analysis data. An `Ember.Service`
+   is a long-lived Ember object that can be made available in different parts of your application.
+ * @example anomaliesApiService: service('services/api/dimensions');
+ */
+export default Service.extend({
+  store: service('store'),
```
[the remark this answers is not in the document]
Comment (newsummit) [reply thirdeye/thirdeye-frontend/app/pods/services/api/dimensions/service.js]: Will do.

File: thirdeye/thirdeye-frontend/app/pods/components/modals/entity-mapping-modal/component.js (hunk 1 above)
Comment (newsummit) [reply thirdeye/thirdeye-frontend/app/pods/components/modals/entity-mapping-modal/component.js]: Ah, yes. Sorry for lack of context. This is related to a query that was failing in the backend and we no longer need: `.../services/all`. After conversations, we don't want to display this in the entity mapping modal dropdown any longer. I can safely just remove it.

**Pull request merged**